### DISCUSSION TITLE
Stability on older smartphones + hash size

### DISCUSSION THREAD

**Discussion by Mercutio1243** [ArjanAswal/Stockfish#38] [Q&A] [answered]
Does anybody has any experience in running the Stockfish package on older smartphones? I am getting some crash reports for the Galaxy A7 (2018) and C1 Max smartphone, which I cannot fully reproduce with my own setup. Could it be that older smartphones are not fully capable of running the package with sufficient stability? Any experience with this?

Related question: how do you decide on the hash size per smartphone? ("setoption name Hash value XXX"). Which sizes are appropriate depending on the set-up and how to you set it programmatically? Does it make sense to check the available RAM size?

Many thanks in advance!

**Comment by veloce** [comment]
I was able to run it on a Nokia 5 (2017) without problem.

You definitely need to check the available RAM. For instance, lichess mobile app allows a maximum of 1/10th of the total memory.

**Comment by Mercutio1243** [comment, marked as answer]
After a year, I actually need would like to come back to this topic: Does anyone have some more experience on whether I need to worry about the hash size to avoid errors / crashes?

I am currently settting hash size for stockfish at (restrictive?) 500mb. I am not fully sure whether this is even needed. Do I need to manage hash size proactively or can I simply keep the default setting (which I understand to be 16 MB?)?

Does anyone have a good approach on this?

Edit: I found this helpful explanation on the hash size. This leads me to simply stay at 500MB, which seems sufficient and has not led to any major breakdowns in the past https://www.reddit.com/r/Chess_/comments/1jsdvbv/understanding_the_ideal_hash_size_for_stockfish_a/
